### ISSUE TITLE
Editor: unhandled error after clicking/holding mouse1 + mouse2 on a hitcircle

### ISSUE THREAD

**Issue by mouzedrift** [ppy/osu#7838]
i couldnt reproduce this since the last time.

**osu!lazer version:**
commit 2fa7143cda91cdb484a87141f042bf7e4ebbc103

**Logs:** 
[runtime.log](https://github.com/ppy/osu/files/4204611/runtime.log)

**Comment by peppy** [comment]
Removing the assert results in the following issue:

- Mouse down left
- Mouse down right
- Mouse up left
- Mouse up right

Selection is lost when it shouldn't be.

Refactoring this whole code would be best. It's very hard to follow right now.

**Comment by peppy** [comment]
This was fixed somewhere along the way.
